src/ripple/server/impl/Role.cpp

```diff
@@ -38,29 +38,28 @@ passwordUnrequiredOrSentCorrect (HTTP::Port const& port,
 }
 
 bool
-ipAllowed (beast::IP::Endpoint const& remoteIp,
-           std::vector<beast::IP::Endpoint> const& adminAllow)
+ipAllowed (beast::IP::Address const& remoteIp,
+           std::vector<beast::IP::Address> const& adminIp)
 {
-    return std::find(adminAllow.begin(), adminAllow.end(),
-                     remoteIp.at_port(0)) != adminAllow.end();
+    return ! adminIp.empty () && std::find_if (adminIp.begin (), adminIp.end (),
+        [&remoteIp](beast::IP::Address const& ip) { return ip.is_any () ||
```

Comment:
Why is the `! adminIp.empty ()` needed? It looks like it works the same without it.